test/dates/io.jl

```diff
@@ -77,7 +94,7 @@ c3 = "96:1:01"
 @test Dates.DateTime(c3,f) + Dates.Year(1900) + Dates.Day(14) == dt
 @test Dates.format(dt-Dates.Day(14),"yy:m:dd") == c3
 c4 = "1996:15:01 # random comment"
-@test Dates.DateTime(c4,f) == dt
+@test_throws ArgumentError Dates.DateTime(c4,f)
```

Comment:
is this breaking then, or was allowing comments before unintentional? can that capability be recreated now with a fancier date format if someone needs it?